libres/lib/enkf/obs_data.cpp

```diff
@@ -361,36 +361,35 @@ void obs_data_free(obs_data_type *obs_data) {
     free(obs_data);
 }
 
-matrix_type *obs_data_allocE(const obs_data_type *obs_data, rng_type *rng,
-                             int active_ens_size) {
-    matrix_type *E;
+Eigen::MatrixXd obs_data_allocE(const obs_data_type *obs_data, rng_type *rng,
```

Comment:
The C code with fully manual memory management has had a quite strong which goes like this: "Any function with `alloc` in the name creates a resource which calling scope must discard when it is no longer in use". When this function now returns an object with automatic storage which does no longer require manual clean up I would recommend renaming the function to e.g. `obs_data_makeE()` - the `make` name is quite in line with standard C++ and functions like `make_pair, make_shared` and `make_unique` which all create and return a value object.

Reply:
👍